models/DataObject/AbstractObject.php

```diff
@@ -490,23 +490,9 @@ public static function getList($config = [])
 
     /**
      * @internal
-     *
-     * @param AbstractObject $object
-     *
-     * @return bool
      */
-    protected static function typeMatch(AbstractObject $object)
+    protected static function typeMatch(AbstractObject $object): bool
```

Comment:
Can we also add a simple tests in `ObjectTest`?

```php
    /**
     * Verifies that loading only Concrete object from Concrete::getById().
     */
    public function testConcreteLoading()
    {
        $concreteObject = TestHelper::createEmptyObject(type: DataObject\Concrete::class);
        $loadedConcrete = DataObject\Concrete::getById($concreteObject->getId(), ['force' => true]);

        $this->assertIsObject($loadedConcrete, 'Loaded Concrete should be an object.');

        $nonConcreteObject = TestHelper::createObjectFolder();
        $loadedNonConcrete = DataObject\Concrete::getById($nonConcreteObject->getId(), ['force' => true]);
        
        $this->assertNull($loadedNonConcrete, 'Loaded Concrete should be null.');
    }
```

Reply:
@dvesh3 I added the test https://github.com/pimcore/pimcore/pull/13548/commits/5a2bf5788e50f7417400a8860b48ba46f9de61f3